src/sage/data_structures/binary_list.pyx

```diff
@@ -0,0 +1,118 @@
+r"""
+A data structure to store lists of integer pairs of large size.
+"""
+
+# ****************************************************************************
+#       Copyright (C) 2022 Jonathan Kliem <jonathan.kliem@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+from cysignals.memory cimport check_reallocarray, check_allocarray, sig_free
+from sage.rings.integer cimport smallInteger
+
+# Should be a power of two.
+# Should be neither exposed nor modified.
+cdef size_t length_per_list = 16348
+
+cdef class BinaryList:
+    def __dealloc__(self):
```

Comment:
I think `BinaryList` is a rather unusual name for this

Reply:
Resolved in 1ef47d116cfb9fb2ec67caebe2235c47eb61c100